security-questionnaire.md

```diff
@@ -2,88 +2,88 @@
 
 ## 07. Do the features in your specification expose information about the underlying platform to origins?
 
-
+No. The API does not provide any platform-specific details to origins.
 
 ## 08. Does this specification allow an origin to send data to the underlying platform?
 
-
+No. The API strictly enables storage and retrieval of files identified by hashes without direct interaction with the underlying platform.
 
 ## 09. Do features in this specification enable access to device sensors?
 
-
+No.
 
 ## 10. Do features in this specification enable new script execution/loading mechanisms?
 
-
+No.
 
 ## 11. Do features in this specification allow an origin to access other devices?
 
-
+No.
 
 ## 12. Do features in this specification allow an origin some measure of control over a user agent's native UI?
 
-
+No. The user agent remains in full control, particularly for displaying permission prompts.
 
 ## 13. What temporary identifiers do the features in this specification create or expose to the web?
 
-
+None. File access is based solely on static hashes, which are not session-specific identifiers.
 
 ## 14. How does this specification distinguish between behavior in first-party and third-party contexts?
 
-
+The permission model ensures that file access is user-controlled, regardless of the context. Explicit user consent is required for cross-origin access.
 
 ## 15. How do the features in this specification work in the context of a browser’s Private Browsing or Incognito mode?
 
-
+Files stored in COS are not accessible in Private Browsing or Incognito mode unless explicitly allowed by the user agent. Data stored during such sessions is not retained.
 
 ## 16. Does this specification have both "Security Considerations" and "Privacy Considerations" sections?
 
-
+Yes. The specification includes detailed sections addressing security and privacy implications, including hashing, user consent, and eviction policies.
 
 ## 17. Do features in your specification enable origins to downgrade default security protections?
 
+No. The API requires explicit user consent for all access, ensuring that default security protections remain intact.
 
+## 18. What happens when a document that uses your feature is kept alive in BFCache?
 
-## 18. What happens when a document that uses your feature is kept alive in BFCache
-
-
+The BFCache behavior should be aligned with user agent policies. File access requests may trigger re-validation if necessary.
 
 ## 19. What happens when a document that uses your feature gets disconnected?
 
-
+The file access operation will terminate, and any pending storage or retrieval will fail gracefully with appropriate errors.
 
 ## 20. Does your spec define when and how new kinds of errors should be raised?
 
-
+Yes. The specification defines specific errors such as `NotAllowedError` (for denied permissions) and `NotFoundError` (for unavailable files).
```

Comment:
These are not really "new kinds of errors."

Reply:
Yepp, made this a "No".